### PR TITLE
Adopt the .go-version approach for toolchain versioning

### PR DESCRIPTION
**Pull request by henrybear327**
Reference: https://github.com/etcd-io/bbolt/pull/713#issuecomment-2037719477

### CODE REVIEW

**Comment by henrybear327** [comment]
Dependent on PR https://github.com/etcd-io/bbolt/pull/713.

We will need to rebase it after merging https://github.com/etcd-io/bbolt/pull/713 and then the `go mod` will be correctly set to 1.21.9.

File: .go-version [hunk 1]
```diff
@@ -0,0 +1 @@
+1.21.9
```
Comment (ahrtr) [inline .go-version:1]: Please rollback this change, and let's update & discuss it under https://github.com/etcd-io/bbolt/pull/713

Comment (henrybear327) [reply .go-version:1]: Just to clarify, does the rollback of this change mean 
- to include the code changes made here with PR #713? 
Or 
- just to change the go version here?

Comment (ahrtr) [reply .go-version:1]: The latter. One PR is supposed to only do one thing, so let's do not update the go version in this PR.

Comment (henrybear327) [reply .go-version]: Fixed according to feedback! Thank you for clarifying! :)

**Review by ahrtr — APPROVED** [review]

**Pull request merged**